src/pystatis/table.py

```diff
@@ -104,88 +125,103 @@ def get_data(
 
         Args:
             data (pd.DataFrame): A pandas dataframe created from raw_data
             db_name (str): The name of the database.
+            language (str): The requested language. One of "de" or "en".
 
         Returns:
             pd.DataFrame: Formatted dataframe that omits all unnecessary Code columns
             and includes informative columns names
         """
         match db_name:
             case "genesis":
-                pretty_data = Table.parse_genesis_table(data)
+                pretty_data = Table.parse_genesis_table(data, language)
             case "zensus":
-                pretty_data = Table.parse_zensus_table(data)
+                pretty_data = Table.parse_zensus_table(data, language)
             case "regio":
-                pretty_data = Table.parse_regio_table(data)
+                pretty_data = Table.parse_regio_table(data, language)
             case _:
                 pretty_data = data
 
         return pretty_data
 
     @staticmethod
-    def parse_genesis_table(data: pd.DataFrame) -> pd.DataFrame:
+    def parse_genesis_table(data: pd.DataFrame, language: str) -> pd.DataFrame:
         """Parse GENESIS table ffcsv format into a more readable format"""
-        # Extracts time column with name from first element of Zeit_Label column
-        time = pd.DataFrame({data["Zeit_Label"].iloc[0]: data["Zeit"]})
 
-        # Extracts new column names from first values of the Merkmal_Label columns
-        # and assigns these to the relevant attribute columns (Auspraegung_Label)
-        attributes = data.filter(like="Auspraegung_Label")
-        attributes.columns = data.filter(like="Merkmal_Label").iloc[0].tolist()
+        column_name_dict = COLUMN_NAME_DICT["genesis-regio"][language]
+
+        # Extracts time column with name from last element of Zeit_Label column
+        time = pd.DataFrame({data[column_name_dict["time_label"]].iloc[-1]: data[column_name_dict["time"]]})
+
+        # Extracts new column names from last values of the variable label columns
+        # and assigns these to the relevant attribute columns (variable level)
+        attributes = data.filter(like=column_name_dict["value_label"])
+        attributes.columns = data.filter(like=column_name_dict["variable_label"]).iloc[-1].tolist()
 
         # Selects all columns containing the values
         values = data.filter(like="__")
 
         # Given a name like BEV036__Bevoelkerung_in_Hauptwohnsitzhaushalten__1000
-        # extracts the label and the unit and omit the code
-        values.columns = [name.split("__", maxsplit=1)[1] for name in values.columns]
+        # extracts the label and the unit and omits the code
+        values.columns = [re.split(r"_{2,}", name, maxsplit=1)[1] for name in values.columns]
 
-        pretty_data = pd.concat([time, attributes, values], axis=1)
+        pretty_data = pd.concat([time, attributes, values], axis=1).dropna(axis=0, how="all")
```

Comment:
Not sure whether we actually want dropna, as NaN's contain a lot of "irregular" but relevant information (see other open issues). At first glance, nothing else to complain about :)

Reply:
This only drops rows containing only NaNs

Reply:
This  was added by @PiaSchroeder , I will take a look at the reason for this.

Reply:
should make sense I mean if there is no information at all why keep it but yeah would be nice to document an evas number where this actually happens and why

Reply:
The behaviour is IMO problematic, as all values with some sort of quality sign (or other issues) would get removed - and from my limited experience, this behaviour is quite common. I would therefore leave removing NaNs up the the user and would propose to just assess proper handling of the NaNs.

Reply:
But why are quality signs removed when dropping all NANs? If there is a quality sign then the row cannot be only NaNs...

Reply:
Sorry, was not reading carefully enough and overlooked "all". Resolving this conversation